src/units/drawer.cpp

```diff
@@ -462,6 +469,45 @@ void unit_drawer::redraw_unit(const unit& u) const
 		halo_man.set_location(ac.unit_halo_, halo_x, halo_y);
 	}
 
+	const std::vector<std::string> halos_abilities = u.halo_abilities();
+	bool has_abil_halo = !ac.abil_halos_.empty();
```

Comment:
Please change this to `!ac.abil_halos_.empty() && ac.abil_halos_.front()->valid();`

Reply:
must be also remove https://github.com/wesnoth/wesnoth/commit/e64453540085014bc53651ad64b311fd61d641de?

Reply:
done with https://github.com/wesnoth/wesnoth/pull/7856/commits/0cd02d834385233f8caf3920756116467150987a